### PR TITLE
ISSUE-486 Fix deadlocks in storage layer during concurrent calls to SR

### PR DESCRIPTION
**Pull request by raju-saravanan**
In case of code path for adding schema metadata, we first do a get(schemaMetadataName), if the get returns a null then we add the new schemaMetadata else we return the schemaMetadataId from the get() call. The current transaction isolation level in SR is SERIALIZABLE, in this case, the get() call gets a shared lock and the add() call get an exclusive lock within a transaction. If two threads are invoking this API we might run into a situation where thread1 has obtained a shared lock and thread2 has also obtained the shared lock but thread1 can't get an exclusive lock because of a shared lock held by thread2 and thread2 can't get an exclusive lock because of a shared lock held by thread1. So, we run into a deadlock. The solution here would be to reduce the transaction isolation level to REPETABLE_READS for MySQL and POSTGRES and since ORACLE doesn't use this locking mechanism to in case of SERIALIZABLE, we can retain the current transaction isolation level for ORACLE.

### CODE REVIEW

File: conf/registry.yaml [hunk 1]
```diff
@@ -64,15 +64,15 @@ fileStorageConfiguration:
...
- providerClass: "com.hortonworks.registries.storage.impl.jdbc.JdbcStorageManager"
- properties:
-   db.type: "mysql"
-   queryTimeoutInSecs: 30
-   db.properties:
-     dataSourceClassName: "com.mysql.jdbc.jdbc2.optional.MysqlDataSource"
-     dataSource.url: "jdbc:mysql://localhost/schema_registry"
-     dataSource.user: "registry_user"
-     dataSource.password: "registry_password"
+  providerClass: "com.hortonworks.registries.storage.impl.jdbc.JdbcStorageManager"
```
Comment (priyank5485) [inline conf/registry.yaml]: Any reason for changing mysql to postrgre? Comment still says MySQL.

File: registry-dist/VERSION [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#Created by build system. Do not modify
```
Comment (priyank5485) [inline registry-dist/VERSION]: Remove this file?

File: schema-registry/rest-service/pom.xml [hunk 3]
```diff
@@ -44,6 +44,18 @@
           </exclusion>
         </exclusions>
       </dependency>
+      <!-- https://mvnrepository.com/artifact/mysql/mysql-connector-java -->
+      <dependency>
```
[the remark this answers is not in the document]
Comment (satishd) [reply schema-registry/rest-service/pom.xml]: Right, this dependency should be removed. There was a licensing issue to package with registry distribution. bootstrap-storage.sh downloads the required jar and puts into respective lib dir.

**Comment by raju-saravanan** [comment]
@satishd  and @priyank5485: This PR is a WIP. I simply created this so that one of the break fix engineers could test this further, the PR in itself is not complete.

**Comment by satishd** [comment]
>@satishd and @priyank5485: This PR is a WIP. I simply created this so that one of the break fix engineers could test this further, the PR in itself is not complete.

@raju-saravanan Sure, I will hold my review till you confirm. 👍

**Comment by raju-saravanan** [comment]
@satishd  and @priyank5485: Can you guys review this PR?

**Comment by satishd** [comment]
@raju-saravanan Can you rebase/resolve the conflicts?

**Comment by raju-saravanan** [comment]
@satishd : Rebased

File: storage/core/src/main/java/com/hortonworks/registries/storage/transaction/TransactionEventListener.java [hunk 4]
```diff
@@ -90,18 +90,12 @@ public void onEvent(RequestEvent event) {
                     isTransactionActive = true;
                 }
             } else if (eventType == RequestEvent.Type.RESP_FILTERS_START) {
-                // not supporting transactions to filters
```
Comment (satishd) [inline storage/core/src/main/java/com/hortonworks/registries/storage/transaction/TransactionEventListener.java]: We may soon forget why this change was done. :) 
You can put a comment on why this change is done.

Comment (raju-saravanan) [reply storage/core/src/main/java/com/hortonworks/registries/storage/transaction/TransactionEventListener.java]: @satishd  Done

**Comment by priyank5485** [comment]
+1
